sdcm/remote.py

```diff
@@ -964,3 +973,501 @@ def run_output_check(self, cmd, timeout=None, ignore_status=False,
+        self.known_hosts_file = tempfile.mkstemp()[1]
+        config = Config(overrides={
+                        'load_ssh_config': False,
+                        'UserKnownHostsFile': self.known_hosts_file,
+                        'ServerAliveInterval': 300,
+                        'StrictHostKeyChecking': 'no'})
+        self.connection = Connection(
+            self.hostname,
+            self.user,
+            port=self.port,
+            config=config,
+            connect_timeout=self.connect_timeout,
+            connect_kwargs={
+                'key_filename': os.path.expanduser(self.key_file),
+            })
+
+    def __str__(self):
+        return 'RemoteFabric [%s@%s]' % (self.user, self.hostname)
+
+    def ssh_debug_cmd(self):
+        if self.key_file:
+            return "SSH access -> 'ssh -i %s %s@%s'" % (self.key_file,
+                                                        self.user,
+                                                        self.hostname)
+        else:
+            return "SSH access -> 'ssh %s@%s'" % (self.user,
+                                                  self.hostname)
+
+    def run(self, cmd, timeout=None, ignore_status=False,
+            connect_timeout=300, options='', verbose=True,
+            args=None, log_file=None, watch_stdout_pattern=None, retry=0):
+
+        self.connection.connect_timeout = connect_timeout
+
+        for i in range(retry + 1):
+            try:
+                if verbose:
+                    self.log.debug('Run command {}'.format(cmd))
+                output_watcher = OutputWatcher(self)
+                start_time = time.time()
+                if log_file:
+                    with open(log_file, 'a+') as fd_logfile:
+                        result = self.connection.run(cmd, warn=ignore_status,
+                                                     encoding='utf-8', hide=True,
+                                                     out_stream=fd_logfile,
+                                                     err_stream=fd_logfile,
+                                                     watchers=[output_watcher])
+                else:
+                    result = self.connection.run(cmd, warn=ignore_status,
+                                                 encoding='utf-8',
+                                                 hide=True,
+                                                 watchers=[output_watcher])
+
+                setattr(result, 'duration', time.time() - start_time)
+                setattr(result, 'exit_status', result.exited)
+
+            except Exception as details:
+                if i == retry:
+                    self.log.error('Error executing command: {cmd}\ndetails: {details}'.format(**locals()))
+                    raise
+        if verbose:
+            self.log.debug('Command {} finished with status {}'.format(result.command, result.exited))
+        #     self.log.debug('STDOUT: {}'.format(result.stdout.encode('utf-8')))
```

Comment:
if the watcher is print the  output, should the verbose param passed to it, to decide what todo with the output ?

Reply:
Thanks. This make sense. Adding verbose parameter